planner/core/find_best_task.go

```diff
@@ -1075,6 +1099,155 @@ func (ds *DataSource) convertToTableScan(prop *property.PhysicalProperty, candid
 	return task, nil
 }
 
+func (ds *DataSource) convertToPointGet(prop *property.PhysicalProperty, candidate *candidatePath) (task task, err error) {
+	if !prop.IsEmpty() && !candidate.isMatchProp {
+		return invalidTask, nil
+	}
+	if prop.TaskTp == property.CopDoubleReadTaskType && candidate.isSingleScan ||
+		prop.TaskTp == property.CopSingleReadTaskType && !candidate.isSingleScan {
+		return invalidTask, nil
+	}
+	if candidate.path.StoreType == kv.TiFlash {
+		return invalidTask, nil
+	}
+
+	pointGetPlan := PointGetPlan{
+		ctx:          ds.ctx,
+		schema:       ds.schema.Clone(),
+		dbName:       ds.DBName.L,
+		TblInfo:      ds.TableInfo(),
+		outputNames:  ds.OutputNames(),
+		LockWaitTime: ds.ctx.GetSessionVars().LockWaitTimeout,
+	}.Init(ds.ctx, ds.stats.ScaleByExpectCnt(float64(len(candidate.path.Ranges))), ds.blockOffset)
+	var partitionInfo *model.PartitionDefinition
+	if ds.isPartition {
+		if pi := ds.tableInfo.GetPartitionInfo(); pi != nil {
+			for _, def := range pi.Definitions {
+				if def.ID == ds.physicalTableID {
+					partitionInfo = &def
+				}
+			}
+		}
+	}
+	rTsk := &rootTask{p: pointGetPlan}
+	var cost float64
+	if candidate.path.IsTablePath {
+		pointGetPlan.Handle = candidate.path.Ranges[0].LowVal[0].GetInt64()
+		pointGetPlan.UnsignedHandle = mysql.HasUnsignedFlag(ds.getHandleCol().RetType.Flag)
+		pointGetPlan.PartitionInfo = partitionInfo
+		cost = pointGetPlan.GetCost(ds.TblCols)
+		// Add filter condition to table plan now.
+		if len(candidate.path.TableFilters) > 0 {
+			sessVars := ds.ctx.GetSessionVars()
+			cost += pointGetPlan.stats.RowCount * sessVars.CPUFactor
+			sel := PhysicalSelection{
+				Conditions: candidate.path.TableFilters,
+			}.Init(ds.ctx, ds.stats.ScaleByExpectCnt(prop.ExpectedCnt), ds.blockOffset)
+			sel.SetChildren(pointGetPlan)
+			rTsk.p = sel
+		}
+	} else {
+		if !candidate.path.Index.Unique ||
+			len(candidate.path.Ranges[0].LowVal) != len(candidate.path.Index.Columns) ||
+			candidate.path.Index.HasPrefixIndex() {
+			return invalidTask, nil
+		}
+		pointGetPlan.IndexInfo = candidate.path.Index
+		pointGetPlan.IndexValues = candidate.path.Ranges[0].LowVal
+		pointGetPlan.PartitionInfo = partitionInfo
+		if candidate.isSingleScan {
+			cost = pointGetPlan.GetCost(candidate.path.IdxCols)
+		} else {
+			cost = pointGetPlan.GetCost(ds.TblCols)
+		}
```

Comment:
Just wondering, how does `PointGet` work when the columns needed are not in the unique index?

```
mysql> create table tbl(a int, b int, unique key(b));
Query OK, 0 rows affected (0.01 sec)

mysql> explain select * from tbl where b = 1;
+-------------+---------+------+--------------------+
| id          | estRows | task | operator info      |
+-------------+---------+------+--------------------+
| Point_Get_1 | 1.00    | root | table:tbl, index:b |
+-------------+---------+------+--------------------+
1 row in set (0.01 sec)

mysql> explain select a from tbl where b = 1;
+-------------+---------+------+--------------------+
| id          | estRows | task | operator info      |
+-------------+---------+------+--------------------+
| Point_Get_1 | 1.00    | root | table:tbl, index:b |
+-------------+---------+------+--------------------+
```

Reply:
Point get can't work in the none-unique index. Because the key of the none-unique index is `${index_value}_${handle}`, we can't construct the keys to call the RPC "KVGet".

Reply:
In point-get executor, it realized double read logic.

Reply:
PointGet can do double read itself.